File: frontend/src/stores/session.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import api from '../api/client'

export interface Session {
  id: number
  sessionId: string
  senderCompId: string
  targetCompId: string
  status: string
  port?: number
  createdAt: string
  updatedAt: string
}

export const useSessionStore = defineStore('session', () => {
  const sessions = ref<Session[]>([])

  async function fetchSessions() {
    const response = await api.get('/sessions')
    sessions.value = response.data
  }

  return { sessions, fetchSessions }
})

File: src/main/java/com/fixsimulator/domain/session/Session.java
package com.fixsimulator.domain.session;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "sessions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Session {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "session_id", unique = true, nullable = false)
    private String sessionId;

    @Column(name = "sender_comp_id", nullable = false)
    private String senderCompId;

    @Column(name = "target_comp_id", nullable = false)
    private String targetCompId;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private SessionStatus status;

    @Column(name = "port")
    private Integer port;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
